pipeline/data/importers/mono/hplt.py

```diff
@@ -0,0 +1,219 @@
+import json
+import random
+from contextlib import ExitStack
+from dataclasses import dataclass
+from pathlib import Path
+
+from pipeline.common.datasets import CountingStep, FilteringStep, Statistics
+from pipeline.common.downloads import location_exists, read_lines, write_lines
+from pipeline.common.logging import get_logger
+from pipeline.common.memory import log_memory
+
+logger = get_logger(__name__)
+
+random.seed(38947598475)
+
+
+@dataclass
+class HPLTDocument:
+    """
+    A structured type for the HPLT document entry in a jsonl file.
+    https://hplt-project.org/datasets/v1.2
+
+    Usage:
+        doc = Document(**json)
+    """
+
+    id: int
+    document_lang: str
+    # The list of scores for each line.
+    scores: list[float]
+    # The detected language for the sentence.pl
+    langs: list[str]
+    # All of the text, split by newlines.
+    text: str
+    # Where this URL was scraped: https://hip.univ-orleans.fr/ipac20/ipac.jsp?session=166R9L62F6723.6585606&profile=scd&menu=search&ts=1660916246772
+    url: str
+    # e.g. "cc40"
+    collection: str
+
+    def __post_init__(self):
+        # The sentences in the text, which were separated by newliens.
+        self.lines = self.text.split("\n")
+
+
+@dataclass
+class FilteringStatistics(Statistics):
+    """
+    Gather statistics about the filtering process.
+    """
+
+    shards: FilteringStep
+    visited_lines: FilteringStep
+    document_count: CountingStep
+    final_lines: CountingStep
+
+    def __init__(self, dataset_path: Path) -> None:
+        super().__init__(dataset_path)
+        self.shards = FilteringStep(
+            dataset_path,
+            "How many shards were sampled from. Each shard contains a subset of the "
+            "total datasets available.",
+        )
+        self.visited_lines = FilteringStep(
+            dataset_path,
+            "How many lines were visited and kept from the HPLT documents.",
+        )
+        self.document_count = CountingStep(
+            dataset_path,
+            "How many documents were visited. This can help represent data diversity.",
+        )
+        self.final_lines = CountingStep(
+            dataset_path,
+            "How many lines were actually written. Smaller lines will be combined together.",
```

Comment:
What do you mean by "lines will be combined together"? I don't think we should touch the dataset structure.